src/backend/distributed/executor/local_executor.c

```diff
@@ -124,53 +136,158 @@ ExecuteLocalTaskList(CitusScanState *scanState, List *taskList)
 {
 	EState *executorState = ScanStateGetExecutorState(scanState);
 	ParamListInfo paramListInfo = copyParamList(executorState->es_param_list_info);
-	int numParams = 0;
-	Oid *parameterTypes = NULL;
 	ListCell *taskCell = NULL;
 	uint64 totalRowsProcessed = 0;
+	Job *workerJob = scanState->distributedPlan->workerJob;
 
-	if (paramListInfo != NULL)
+	foreach(taskCell, taskList)
```

Comment:
I thought we'd be  excluding the  changes related  to local execution from this PR. Can't we simply call `TaskQueryString`  followed by `ParseQueryString ` in this PR, and implement local-fast-path specific  ones in #3337?

The idea that I thought we'd be to generalize lazy queryString generation in this  PR,  and skip generating the `queryString` in #3337? Or, am I missing something?

Reply:
We've talked with @JelteF in person, decided to go with the above approach to simplify the code reviews. I'll push a commit that does it soon

Reply:
Done